Development

dmontagner opened the pull request:
HealthBot solution to validate SLAs (5-nines and 6-nines) in the network using RPM probes with custom (xmlproxyd) OpenConfig sensor

gvarao commented:
Hi Diogo,

Thanks for contributing. I have some queries and comments related to this pull request.


  *   Is this use case is generic and can be used by majority of the customers?
  *   If Yes, you need to submit only readme in .md file format, rule files as .rule, UDF as .py and playbook as .playbook. And rest of the files are not required i.e. .conf, .yang, .png
  *   If No, You can commit these rules into our internal repo https://svl-ssd-git.juniper.net/iceberg/customer-healthbot-rules/tree/master. There is no need of pull request. You can commit into master.

Regards,
Venkat(GV)
From: Diogo Montagner <notifications@github.com>
Reply to: Juniper/healthbot-rules <reply@reply.github.com>
Date: Monday, 10 February 2020 at 9:25 AM
To: Juniper/healthbot-rules <healthbot-rules@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [Juniper/healthbot-rules] Development (#282)


HealthBot solution to validate SLAs (5-nines and 6-nines) in the network using RPM probes with custom (xmlproxyd) OpenConfig sensor

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/Juniper/healthbot-rules/pull/282

Commit Summary

  *   Added RPM probe solution
  *   updated README.md with screenshots
  *   minor corrections on README.md
  *   updates to documentation

File Changes

  *   A juniper_official/Solutions/RPM-probes/README.md<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-0> (141)
  *   A juniper_official/Solutions/RPM-probes/oc_yang_model/xmlproxyd_rpm.yang<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-1> (115)
  *   A juniper_official/Solutions/RPM-probes/pictures/1.Playbook-deployed-to-PROBES-device-group.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-2> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/2.SLA_is_all_green_as_no_outages_have_occurred.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-3> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/3.Probe3_went_offline_first_SLA_to_breach_is_6_nines.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-4> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/4.Details_of_the_SLA_breach.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-5> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/5.After_just_over_5_minutes_the_5_nines_SLA_gets_breached.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-6> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/6.Topology_traffic_flow.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-7> (0)
  *   A juniper_official/Solutions/RPM-probes/pictures/Example_topology.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-8> (0)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/pe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-9> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/pe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-10> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-11> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-12> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe3.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-13> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe4.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-14> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe5.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-15> (75)
  *   A juniper_official/Solutions/RPM-probes/probes_configuration/probe6.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-16> (75)
  *   A juniper_official/Solutions/RPM-probes/rpm-probe-playbook.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-17> (7)
  *   A juniper_official/Solutions/RPM-probes/rpm-probe-rule.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-18> (196)
  *   A juniper_official/Solutions/RPM-probes/udf/cumm-pkt-loss-from-db.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-19> (33)
  *   A juniper_official/Solutions/RPM-probes/udf/sla.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-20> (7)

Patch Links:

  *   https://github.com/Juniper/healthbot-rules/pull/282.patch
  *   https://github.com/Juniper/healthbot-rules/pull/282.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/Juniper/healthbot-rules/pull/282?email_source=notifications&email_token=AD4JHVDRUZTZ6TYXUR2Q4BLRCDF23A5CNFSM4KSHBEQ2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IMEFEPQ>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD4JHVCIK32OAELTTIKILSTRCDF23ANCNFSM4KSHBEQQ>.

dmontagner commented:
It is generic and a lot of customers will like this.

I will modify the extensions of the file. The .png are referenced in the .md to facilitate the explanation.

Diogo Montagner
APAC CoE

Sent from my iPhone

On 10 Feb 2020, at 15:36, Venkat <notifications@github.com> wrote:

﻿Hi Diogo,

Thanks for contributing. I have some queries and comments related to this pull request.


* Is this use case is generic and can be used by majority of the customers?
* If Yes, you need to submit only readme in .md file format, rule files as .rule, UDF as .py and playbook as .playbook. And rest of the files are not required i.e. .conf, .yang, .png
* If No, You can commit these rules into our internal repo https://svl-ssd-git.juniper.net/iceberg/customer-healthbot-rules/tree/master. There is no need of pull request. You can commit into master.

Regards,
Venkat(GV)
From: Diogo Montagner <notifications@github.com>
Reply to: Juniper/healthbot-rules <reply@reply.github.com>
Date: Monday, 10 February 2020 at 9:25 AM
To: Juniper/healthbot-rules <healthbot-rules@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [Juniper/healthbot-rules] Development (#282)


HealthBot solution to validate SLAs (5-nines and 6-nines) in the network using RPM probes with custom (xmlproxyd) OpenConfig sensor

________________________________
You can view, comment on, or merge this pull request online at:

https://github.com/Juniper/healthbot-rules/pull/282

Commit Summary

* Added RPM probe solution
* updated README.md with screenshots
* minor corrections on README.md
* updates to documentation

File Changes

* A juniper_official/Solutions/RPM-probes/README.md<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-0> (141)
* A juniper_official/Solutions/RPM-probes/oc_yang_model/xmlproxyd_rpm.yang<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-1> (115)
* A juniper_official/Solutions/RPM-probes/pictures/1.Playbook-deployed-to-PROBES-device-group.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-2> (0)
* A juniper_official/Solutions/RPM-probes/pictures/2.SLA_is_all_green_as_no_outages_have_occurred.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-3> (0)
* A juniper_official/Solutions/RPM-probes/pictures/3.Probe3_went_offline_first_SLA_to_breach_is_6_nines.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-4> (0)
* A juniper_official/Solutions/RPM-probes/pictures/4.Details_of_the_SLA_breach.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-5> (0)
* A juniper_official/Solutions/RPM-probes/pictures/5.After_just_over_5_minutes_the_5_nines_SLA_gets_breached.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-6> (0)
* A juniper_official/Solutions/RPM-probes/pictures/6.Topology_traffic_flow.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-7> (0)
* A juniper_official/Solutions/RPM-probes/pictures/Example_topology.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-8> (0)
* A juniper_official/Solutions/RPM-probes/probes_configuration/pe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-9> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/pe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-10> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-11> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-12> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe3.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-13> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe4.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-14> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe5.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-15> (75)
* A juniper_official/Solutions/RPM-probes/probes_configuration/probe6.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-16> (75)
* A juniper_official/Solutions/RPM-probes/rpm-probe-playbook.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-17> (7)
* A juniper_official/Solutions/RPM-probes/rpm-probe-rule.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-18> (196)
* A juniper_official/Solutions/RPM-probes/udf/cumm-pkt-loss-from-db.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-19> (33)
* A juniper_official/Solutions/RPM-probes/udf/sla.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-20> (7)

Patch Links:

* https://github.com/Juniper/healthbot-rules/pull/282.patch
* https://github.com/Juniper/healthbot-rules/pull/282.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/Juniper/healthbot-rules/pull/282?email_source=notifications&email_token=AD4JHVDRUZTZ6TYXUR2Q4BLRCDF23A5CNFSM4KSHBEQ2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IMEFEPQ>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD4JHVCIK32OAELTTIKILSTRCDF23ANCNFSM4KSHBEQQ>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Juniper/healthbot-rules/pull/282?email_source=notifications&email_token=ABFVDWUP325HEE554JA7YPTRCDKTRA5CNFSM4KSHBEQ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELHGJWA#issuecomment-583951576>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABFVDWQSIOVL4FGDG7YU4ULRCDKTRANCNFSM4KSHBEQQ>.

dmontagner commented:
Also, the .yang is required to be loaded in the routers. Otherwise, HB won’t be able to subscribe to the xmlproxyd sensor.


Diogo Montagner
(m) +61.499.075.297
APAC CoE
Juniper Networks

> On 10 Feb 2020, at 15:51, Diogo Montagner <dmontagner@juniper.net> wrote:
> 
> It is generic and a lot of customers will like this.
> 
> I will modify the extensions of the file. The .png are referenced in the .md to facilitate the explanation.
> 
> Diogo Montagner
> APAC CoE
> 
> Sent from my iPhone
> 
>> On 10 Feb 2020, at 15:36, Venkat <notifications@github.com> wrote:
>> 
>> ﻿Hi Diogo, 
>> 
>> Thanks for contributing. I have some queries and comments related to this pull request. 
>> 
>> 
>> * Is this use case is generic and can be used by majority of the customers? 
>> * If Yes, you need to submit only readme in .md file format, rule files as .rule, UDF as .py and playbook as .playbook. And rest of the files are not required i.e. .conf, .yang, .png 
>> * If No, You can commit these rules into our internal repo https://svl-ssd-git.juniper.net/iceberg/customer-healthbot-rules/tree/master. There is no need of pull request. You can commit into master. 
>> 
>> Regards, 
>> Venkat(GV) 
>> From: Diogo Montagner <notifications@github.com> 
>> Reply to: Juniper/healthbot-rules <reply@reply.github.com> 
>> Date: Monday, 10 February 2020 at 9:25 AM 
>> To: Juniper/healthbot-rules <healthbot-rules@noreply.github.com> 
>> Cc: Subscribed <subscribed@noreply.github.com> 
>> Subject: [Juniper/healthbot-rules] Development (#282) 
>> 
>> 
>> HealthBot solution to validate SLAs (5-nines and 6-nines) in the network using RPM probes with custom (xmlproxyd) OpenConfig sensor 
>> 
>> ________________________________ 
>> You can view, comment on, or merge this pull request online at: 
>> 
>> https://github.com/Juniper/healthbot-rules/pull/282 
>> 
>> Commit Summary 
>> 
>> * Added RPM probe solution 
>> * updated README.md with screenshots 
>> * minor corrections on README.md 
>> * updates to documentation 
>> 
>> File Changes 
>> 
>> * A juniper_official/Solutions/RPM-probes/README.md<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-0> (141) 
>> * A juniper_official/Solutions/RPM-probes/oc_yang_model/xmlproxyd_rpm.yang<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-1> (115) 
>> * A juniper_official/Solutions/RPM-probes/pictures/1.Playbook-deployed-to-PROBES-device-group.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-2> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/2.SLA_is_all_green_as_no_outages_have_occurred.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-3> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/3.Probe3_went_offline_first_SLA_to_breach_is_6_nines.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-4> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/4.Details_of_the_SLA_breach.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-5> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/5.After_just_over_5_minutes_the_5_nines_SLA_gets_breached.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-6> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/6.Topology_traffic_flow.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-7> (0) 
>> * A juniper_official/Solutions/RPM-probes/pictures/Example_topology.png<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-8> (0) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/pe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-9> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/pe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-10> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe1.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-11> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe2.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-12> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe3.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-13> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe4.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-14> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe5.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-15> (75) 
>> * A juniper_official/Solutions/RPM-probes/probes_configuration/probe6.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-16> (75) 
>> * A juniper_official/Solutions/RPM-probes/rpm-probe-playbook.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-17> (7) 
>> * A juniper_official/Solutions/RPM-probes/rpm-probe-rule.cnf<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-18> (196) 
>> * A juniper_official/Solutions/RPM-probes/udf/cumm-pkt-loss-from-db.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-19> (33) 
>> * A juniper_official/Solutions/RPM-probes/udf/sla.py<https://github.com/Juniper/healthbot-rules/pull/282/files#diff-20> (7) 
>> 
>> Patch Links: 
>> 
>> * https://github.com/Juniper/healthbot-rules/pull/282.patch 
>> * https://github.com/Juniper/healthbot-rules/pull/282.diff 
>> 
>> — 
>> You are receiving this because you are subscribed to this thread. 
>> Reply to this email directly, view it on GitHub<https://github.com/Juniper/healthbot-rules/pull/282?email_source=notifications&email_token=AD4JHVDRUZTZ6TYXUR2Q4BLRCDF23A5CNFSM4KSHBEQ2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IMEFEPQ>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD4JHVCIK32OAELTTIKILSTRCDF23ANCNFSM4KSHBEQQ>. 
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub, or unsubscribe.
>>
